Re-enable and stabilise system test PythonChannels.PythonChannelTest

**Describe the outcome that is desired.**
In https://github.com/mantidproject/mantid/pull/37364 the `PythonChannels.PythonChannelTest` was disabled after several random failures in the last few days.

This problem is not new: https://github.com/mantidproject/mantid/pull/36326, https://github.com/mantidproject/mantid/pull/36358 and https://github.com/mantidproject/mantid/pull/36362.

**Describe any solutions you are considering**
The aim is to find a permanent solution. The default behaviour of `multiprocessing` will change in Python 3.14, so it would be good to keep that in mind.

**Additional context**
Further information can be found here: https://docs.python.org/3/library/multiprocessing.html

Since no one linked to the actual test failure in this issue or #37364, here is the failure as seen on the build server from [7377](https://builds.mantidproject.org/view/Pull%20Requests/job/pull_requests-conda-linux/7377/)

```
======================================================================
ERROR: test_PythonStdoutChannel_update_instruments (PythonChannels.PythonLoggingTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/jenkins_workdir/workspace/pull_requests-conda-linux/Testing/SystemTests/tests/framework/PythonChannels.py", line 84, in test_PythonStdoutChannel_update_instruments
    self._create_script_and_execute(LoggingClass="PythonStdoutChannel", UpdateInstrumentDefinitions=1)
  File "/jenkins_workdir/workspace/pull_requests-conda-linux/Testing/SystemTests/tests/framework/PythonChannels.py", line 99, in _create_script_and_execute
    result = subprocess.run([sys.executable, TEST_SCRIPT_NAME], timeout=60)  # This will raise subprocess.TimeoutExpired if deadlocke
  File "/jenkins_workdir/workspace/pull_requests-conda-linux/mambaforge/envs/mantid-developer/lib/python3.10/subprocess.py", line 505, in run
    stdout, stderr = process.communicate(input, timeout=timeout)
  File "/jenkins_workdir/workspace/pull_requests-conda-linux/mambaforge/envs/mantid-developer/lib/python3.10/subprocess.py", line 1154, in communicate
    stdout, stderr = self._communicate(input, endtime, timeout)
  File "/jenkins_workdir/workspace/pull_requests-conda-linux/mambaforge/envs/mantid-developer/lib/python3.10/subprocess.py", line 2047, in _communicate
    self.wait(timeout=self._remaining_time(endtime))
  File "/jenkins_workdir/workspace/pull_requests-conda-linux/mambaforge/envs/mantid-developer/lib/python3.10/subprocess.py", line 1209, in wait
    return self._wait(timeout=timeout)
  File "/jenkins_workdir/workspace/pull_requests-conda-linux/mambaforge/envs/mantid-developer/lib/python3.10/subprocess.py", line 1951, in _wait
    raise TimeoutExpired(self.args, timeout)
subprocess.TimeoutExpired: Command '['/jenkins_workdir/workspace/pull_requests-conda-linux/mambaforge/envs/mantid-developer/bin/python3', 'pythonloggingtestscript.py']' timed out after 59.999970824457705 seconds

----------------------------------------------------------------------
Ran 4 tests in 99.908s

FAILED (errors=1)
Traceback (most recent call last):
  File "/tmp/tmpqlr5dqn1", line 12, in <module>
    systest.execute()
  File "/jenkins_workdir/workspace/pull_requests-conda-linux/Testing/SystemTests/lib/systemtests/systemtesting.py", line 248, in execute
    self.runTest()
  File "/jenkins_workdir/workspace/pull_requests-conda-linux/Testing/SystemTests/tests/framework/PythonChannels.py", line 119, in runTest
    self.assertTrue(runner.run(suite).wasSuccessful())
  File "/jenkins_workdir/workspace/pull_requests-conda-linux/mambaforge/envs/mantid-developer/lib/python3.10/unittest/case.py", line 687, in assertTrue
    raise self.failureException(msg)
AssertionError: False is not true
```

> This problem is not new: https://github.com/mantidproject/mantid/pull/36326, https://github.com/mantidproject/mantid/pull/36358 and https://github.com/mantidproject/mantid/pull/36362.

It is unrelated to the issue that OSX was having with this test.

I looked through 3 months of builds on `pull_requests-conda-linux` and the earliest instance of this test failing was [6932](https://builds.mantidproject.org/view/Pull%20Requests/job/pull_requests-conda-linux/6932/) (1 month and 8 days ago), after which it was failing somewhat often. So this is actually a new issue that didn't exist before that.

I don't know what changed that could be causing the issue and I am unable to reproduce the issue locally.

My suggestion is to remove the test completely. The code comments in the actual code fixes is hopefully enough to prevent someone accidentally removing them and the users that pick up the issue originally will quickly find them again if they come back.

But if you do feel like we should try to stabilize the tests have 2 thoughts.
1. You can remove the [10 second sleep](https://github.com/mantidproject/mantid/blob/29f93dd7966217ebf3322ab890ffa4be9094b6ec/Testing/SystemTests/tests/framework/PythonChannels.py#L51) as that was only included to get around a bug that was later fixed by #36798
2. Increase the [timeout](https://github.com/mantidproject/mantid/blob/29f93dd7966217ebf3322ab890ffa4be9094b6ec/Testing/SystemTests/tests/framework/PythonChannels.py#L98), I am only guessing that something is slowing down this test to the point that the test fails, if you increase it to 600 seconds and it still randomly fails then there must be some other issue.

--- a/Testing/SystemTests/tests/framework/PythonChannels.py
+++ b/Testing/SystemTests/tests/framework/PythonChannels.py
@@ -48,7 +48,6 @@ ConfigService["CheckMantidVersion.OnStartup"] = "{CheckMantidVersion}"
 ConfigService["UpdateInstrumentDefinitions.OnStartup"] = "{UpdateInstrumentDefinitions}"
 
 import mantid.simpleapi
-time.sleep(10)
 """
 
 TEST_SCRIPT_NAME = "pythonloggingtestscript.py"
@@ -95,7 +94,7 @@ class PythonLoggingTests(unittest.TestCase):
                 )
             )
 
-        result = subprocess.run([sys.executable, TEST_SCRIPT_NAME], timeout=60)  # This will raise subprocess.TimeoutExpired if deadlocke
+        result = subprocess.run([sys.executable, TEST_SCRIPT_NAME], timeout=600)  # This will raise subprocess.TimeoutExpired if deadlocke
         result.check_returncode()
 
 
@@ -105,8 +104,8 @@ class PythonChannelTest(systemtesting.MantidSystemTest):
         os.remove(TEST_SCRIPT_NAME)
 
     def skipTests(self):
-        # temporarily skip this test for all platforms
-        return True
+        # skip if OSX because multiprocessing uses `spawn` instead of `fork` which causes this test to fail
+        return platform.system() == "Darwin"
 
     def runTest(self):
         self._server = Process(target=run_server)

--- a/Testing/SystemTests/tests/framework/PythonChannels.py
+++ b/Testing/SystemTests/tests/framework/PythonChannels.py
@@ -26,6 +26,7 @@ import time
 import unittest
 import subprocess
 import systemtesting
+import platform
 from multiprocessing import Process
 from http.server import HTTPServer, BaseHTTPRequestHandler
 

